pkg/controllers/netpol/network_policy_controller.go

```diff
@@ -1663,12 +1663,88 @@ func (npc *NetworkPolicyController) newNetworkPolicyEventHandler() cache.Resourc
 			npc.OnNetworkPolicyUpdate(newObj)
 		},
 		DeleteFunc: func(obj interface{}) {
-			npc.OnNetworkPolicyUpdate(obj)
+			npc.handleNetworkPolicyDelete(obj)
 
 		},
 	}
 }
 
+func (npc *NetworkPolicyController) handlePodDelete(obj interface{}) {
+	pod, ok := obj.(*api.Pod)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			glog.Errorf("unexpected object type: %v", obj)
+			return
+		}
+		if pod, ok = tombstone.Obj.(*api.Pod); !ok {
+			glog.Errorf("unexpected object type: %v", obj)
+			return
+		}
+	}
+	glog.V(2).Infof("Received pod: %s/%s delete event", pod.Namespace, pod.Name)
+	if !npc.readyForUpdates {
+		glog.V(3).Infof("Skipping pod: %s/%s delete event, controller still performing bootup full-sync", pod.Namespace, pod.Name)
+		return
+	}
+
+	err := npc.Sync()
+	if err != nil {
+		glog.Errorf("Error syncing network policy for pod: %s/%s delete event Error: %s", pod.Namespace, pod.Name, err)
+	}
+}
+
+func (npc *NetworkPolicyController) handleNamespaceDelete(obj interface{}) {
+	namespace, ok := obj.(*api.Namespace)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			glog.Errorf("unexpected object type: %v", obj)
+			return
+		}
+		if namespace, ok = tombstone.Obj.(*api.Namespace); !ok {
+			glog.Errorf("unexpected object type: %v", obj)
+			return
+		}
+	}
+	// namespace (and annotations on it) has no significance in GA ver of network policy
+	if npc.v1NetworkPolicy {
+		return
+	}
+	glog.V(2).Infof("Received namespace: %s delete event", namespace.Name)
+
+	err := npc.Sync()
```

Comment:
Do we not need to check if the npc is ready for updates before we call `Sync` like we do in the other functions?
e.g.
```
if !npc.readyForUpdates
```

Reply:
@aauren thanks for the review. Sorry abouth the dealy. I updated the PR to include suggested change.